reporter/process.go

```diff
@@ -0,0 +1,197 @@
+//
+// Copyright 2022 SkyAPM org
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package reporter
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	commonv3 "skywalking.apache.org/repo/goapi/collect/common/v3"
+)
+
+type ProcessReportStatus int8
+
+const (
+	ProcessLabelKey = "processLabels"
+
+	NotInit ProcessReportStatus = iota
+	Reported
+	Confirmed
+	Closed
+)
+
+var process *processStat
+
+type processStat struct {
+	basePath        string
+	metaFilePath    string
+	confirmFilePath string
+	status          ProcessReportStatus
+	shutdownOnce    sync.Once
+}
+
+func initProcessStat(r *gRPCReporter) *processStat {
+	basePath := path.Join(os.TempDir(), "apache_skywalking", "process", strconv.Itoa(os.Getpid()))
+	metaFilePath := path.Join(basePath, "metadata.properties")
+	confirmFilePath := path.Join(basePath, "metadata-confirm.properties")
+
+	return &processStat{
+		basePath:        basePath,
+		metaFilePath:    metaFilePath,
+		confirmFilePath: confirmFilePath,
+		status:          NotInit,
+	}
+}
+
+// Report the current process metadata to local file
+// using to work with eBPF agent
+func reportProcessIFNeed(r *gRPCReporter) {
```

Comment:
I think we need a very clear option for this new feature. Labels are optional, this should not

Reply:
And please document clearly what file(s) is being created, and which information would be written.
Then when the rover side doc is ready, this should link to there for further information.

Reply:
> I think we need a very clear option for this new feature. Labels are optional, this should not

Yes, this function is not optional, maybe the function name let you feel confused. updated.

Reply:
> And please document clearly what file(s) is being created, and which information would be written. Then when the rover side doc is ready, this should link to there for further information.

Updated. After this merge, I will update the documentation on the rover side and finally link the document there.